### PR TITLE
Update How_to_add_a_new_data_type.md

### PR DESCRIPTION
**Pull request by nlharris**
Added title, made some minor wording/punctuation improvements.

Added FIXME to that places that need work, to make it harder to forget to update them.

Comment on line 417: "From solrconfig.xml you need to modify line 106 to point to the correct directory for the index data of your type." If there's a new release of solrconfig.xml, it might not be line 106 anymore, so you should include that line here (or at least say which variable you need to set/modify). (Similarly for the other line numbers mentioned here--describe what's on the lines. No one is going to think to come and update this document if the SOLR installation is updated and the line numbers change!)

Line 429: Does Solr use the term "ingest"? If not, I would change this to "input" (but if "ingest" is their term for it, then it's ok to leave it).

### CODE REVIEW

File: How_to_add_a_new_data_type.md [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
+# How to add a new data type to KBase
+
+1.  First you must submit documentation about what your data type would be used for, and how it fits into KBase. **You** *must* **wait for approval of this documentation before proceeding to step 2.** This documentation should include at a minimum:
 
     1.  A diagram that shows the biological concepts that your data describes, and how it relates to other biological concepts in KBase. This diagram should not be tied to a specific implementation of the data storage of the type.
 
-    2.  A document that describes in detail all elements of the diagram, what each entity is, why it is important, what it describes/contains and why and what the relationships mean and why they are important. You should include known applications (use cases) that use this data type, how it is used and why.
+    2.  A document that describes in detail all elements of the diagram, what each entity is, why it is important, what it describes/contains and why and what the relationships mean and why they are important. You should include known applications (use cases) that use this data type, showing how it is used and why.
```
Comment (mlhenderson) [inline How_to_add_a_new_data_type.md]: explaining

File: How_to_add_a_new_data_type.md [hunk 2]
```diff
@@ -332,23 +334,23 @@ Standard Input keywords:
...
-> **working\_directory** - where you can write to, if needed
+> **working\_directory** - directory to write to, if needed
 >
-> **shock\_id** - if data started out in shock, you can combine this with the service url to save a reference to the data in the workspace object
+> **shock\_id** - if data started out in shock, you can combine this with the service URL to save a reference to the data in the workspace object
 >
 > **handle\_id** - if data started out in shock and you have a handle reference to this data and the workspace object has a field for saving a handle reference
 >
-> **input\_mapping** - an input mapping that maps input keys
+> **input\_mapping** - an input mapping that maps input keys [FIXME: to what??]
```
Comment (mlhenderson) [inline How_to_add_a_new_data_type.md]: maps input key value pairs to command line arguments and values

File: How_to_add_a_new_data_type.md [hunk 3]
```diff
@@ -380,19 +382,19 @@ Standard input keywords:
 
 > **Landing page / Narrative Visualization Widget**
 >
-> Your data type should have some minimal visualization in both the Narrative interface (so that clicking on the data object in the left-hand data list or dragging and dropping on to the narrative cells opens the viewer) and in the functional-site to show on the landing page for your data. Visualizations should be primarily keyed off of a single typed object defined and registered with the Workspace.
+> Your data type should have some minimal visualization in both the Narrative interface (so that clicking on the data object in the left-hand data list or dragging and dropping onto the Narrative panel opens the viewer) and in the functional site to show on the landing page for your data. Visualizations should be primarily keyed off of a single typed object defined and registered with the Workspace.
```
Comment (mlhenderson) [inline How_to_add_a_new_data_type.md:385]: not sure about this, shouldn't it be a narrative cell?

**Comment by mlhenderson** [comment]
Data ingestion ("ingest") is common terminology, not just solr specific language.  Maybe I'm missing something here.

**Comment by mlhenderson** [comment]
It is my _responsibility_ to keep this document up to date.

**Comment by nlharris** [comment]
> On Apr 16, 2015, at 3:05 PM, Matt Henderson notifications@github.com wrote:
> 
> It is my responsibility to keep this document up to date.

Great! I just figured there were ways to lower the hassle of updating it.

**Comment by mlhenderson** [comment]
I don't think I can reply to a comment in that way directly using github.
Maybe if I strictly used email, but then I can't accept the PR that way.

File: How_to_add_a_new_data_type.md (hunk 3 above)
Comment (nlharris) [inline How_to_add_a_new_data_type.md:385]: Yes--it becomes a Narrative cell when you drag it onto the Narrative panel. "Narrative panel" (or "main Narrative panel") was the terminology we agreed on for this section of the NI. You're not dragging the object onto a cell.
I don't feel strongly about this, though, so if you object to this change, I'm not going to push it.

File: How_to_add_a_new_data_type.md (hunk 1 above)
Comment (nlharris) [inline How_to_add_a_new_data_type.md]: Ok

**Comment by fperez** [comment]
@mlhenderson, since you're on top of this doc, do you think this is ready for merge at this point, or do you guys want further work/discussion before merging?

**Comment by mlhenderson** [comment]
@fperez It is not ready to merge yet.  Let me check with Nomi on the changes I asked for and see if we can fix it so it can be accepted.  Most of her edits are fine, there were a couple that I wanted changed before merging.

**Comment by mlhenderson** [comment]
Thanks Nomi!

**Pull request merged**